src/test/java/seedu/address/storage/JsonDietLahStorageTest.java

```diff
@@ -0,0 +1,110 @@
+    }
+
+    private Path addToTestDataPathIfNotNull(String prefsFileInTestDataFolder) {
+        return prefsFileInTestDataFolder != null
+                ? TEST_DATA_FOLDER.resolve(prefsFileInTestDataFolder)
+                : null;
+    }
+
+    @Test
+    public void read_missingFile_emptyResult() throws Exception {
+        assertFalse(readDietLah("NonExistentFile.json").isPresent());
+    }
+
+    @Test
+    public void read_notJsonFormat_exceptionThrown() {
+        assertThrows(DataConversionException.class, () -> readDietLah("notJsonFormatAddressBook.json"));
+    }
+
+    @Test
+    public void readDietLah_invalidPersonDietLah_throwDataConversionException() {
+        assertThrows(DataConversionException.class, () -> readDietLah("invalidPersonAddressBook.json"));
+    }
+
+    @Test
+    public void readDietLah_invalidAndValidPersonDietLah_throwDataConversionException() {
+        assertThrows(DataConversionException.class, () -> readDietLah("invalidAndValidPersonAddressBook.json"));
+    }
+
+    @Test
+    public void readAndSaveDietLah_allInOrder_success() throws Exception {
+        Path filePath = testFolder.resolve("TempAddressBook.json");
+        DietLah original = getTypicalDietLah();
+        JsonDietLahStorage jsonDietLahStorage = new JsonDietLahStorage(filePath);
+
+        // Save in new file and read back
+        jsonDietLahStorage.saveDietLah(original, filePath);
+        ReadOnlyDietLah readBack = jsonDietLahStorage.readDietLah(filePath).get();
+        assertEquals(original, new DietLah(readBack));
+
+        // Modify data, overwrite exiting file, and read back
+        original.addPerson(HOON);
+        original.removePerson(ALICE);
+        jsonDietLahStorage.saveDietLah(original, filePath);
+        readBack = jsonDietLahStorage.readDietLah(filePath).get();
+        assertEquals(original, new DietLah(readBack));
+
+        // Save and read without specifying file path
+        original.addPerson(IDA);
+        jsonDietLahStorage.saveDietLah(original); // file path not specified
+        readBack = jsonDietLahStorage.readDietLah().get(); // file path not specified
+        assertEquals(original, new DietLah(readBack));
+
+    }
+
+    @Test
+    public void saveAddressBook_nullAddressBook_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> saveDietLah(null, "SomeFile.json"));
+    }
+
+    /**
+     * Saves {@code dietLah} at the specified {@code filePath}.
+     */
+    private void saveDietLah(ReadOnlyDietLah dietLah, String filePath) {
+        try {
+            new JsonDietLahStorage(Paths.get(filePath))
+                    .saveDietLah(dietLah, addToTestDataPathIfNotNull(filePath));
+        } catch (IOException ioe) {
+            throw new AssertionError("There should not be an error writing to the file.", ioe);
+        }
+    }
+
+    @Test
+    public void saveAddressBook_nullFilePath_throwsNullPointerException() {
```

Comment:
I think for some of these methods we have to rename from AddressBook also

Reply:
Ah yes, thanks for pointing it out!